File: code/library/src/main/java/org/fbme/lib/iec61499/fbnetwork/FunctionBlockDeclaration.java
package org.fbme.lib.iec61499.fbnetwork;

import org.fbme.lib.common.Reference;
import org.fbme.lib.iec61499.declarations.FBTypeDeclaration;
import org.fbme.lib.iec61499.declarations.ParameterAssignment;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface FunctionBlockDeclaration extends FunctionBlockDeclarationBase {

    @Nullable FBNetwork getContainer();

    @NotNull Reference<FBTypeDeclaration> getTypeReference();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/fbnetwork/FunctionBlockDeclaration.kt
package org.fbme.lib.iec61499.fbnetwork

import org.fbme.lib.common.Reference
import org.fbme.lib.iec61499.declarations.FBTypeDeclaration

interface FunctionBlockDeclaration : FunctionBlockDeclarationBase {
    var id: String?
    override val container: FBNetwork?
    val typeReference: Reference<FBTypeDeclaration>
    var namespace: String?
}
